### ISSUE TITLE
🐛 Bug Report: [Azure DevOps] Build definition annotation not filtering correctly

### ISSUE THREAD

**Issue by antoniobergas** [backstage/backstage#17151]
### 📜 Description

Hi!

We are using the Azure DevOps plugin with a self-hosted instance. Mostly everything works great and it's super useful for our day by day.

In this days we start having quite a lot mono-repos which means that the plugin shows all the builds linked to the repo.

In theory the annotation build-definition should do the trick in this scenarios but unfortunately is not showing the list of build filtered.

I recheck the configuration lot of times and tried to follow the tradability of the calls but have no luck.

Not 100% sure but looks like there's some bug in the filter params of the call.

### 👍 Expected behavior

After add build-definition to the catalog-info.yaml of a mono-repo the list of builds in Entity Page it's filtered by the name of the pipeline.

### 👎 Actual Behavior with Screenshots

After add the name of the pipeline in the catalog, following the readme of the plugin the list of builds is not filtered properly.
In this scenario we have multiple folders in the same repo, each folder have a azure-pipelines.yml with his own pipeline.
![image](https://user-images.githubusercontent.com/26031467/228522550-26b9c827-c7e0-45ed-bb78-e43827b8b56b.png)
![image](https://user-images.githubusercontent.com/26031467/228522733-ed246a3d-4998-47e0-886d-b985277b377f.png)

### 👟 Reproduction steps

1. Add azure devops plugin having a self-hosted instance configured in the app-config.yml
2. Have a repo withwith multiple folder, with 1 azure-pipelines.yml and 1 pipelines per each yml file
3. Add a catalog with the build-definition annotation with the name of 1 of the pipelines
4. Then you will see that the list of buils in Backstage is not filtered

### 📃 Provide the context for the Bug.

Provide a list of builds filtered by "folder/app/component" from the same repo in the CI/CD Entity page tab

### 🖥️ Your Environment

OS:   Linux 5.10.16.3-microsoft-standard-WSL2 - linux/x64
node: v18.9.0
yarn: 1.22.19
cli:  0.22.5 (installed)
backstage:  1.12.1

Dependencies:
  @backstage/app-defaults                          1.2.1
  @backstage/backend-app-api                       0.4.1
  @backstage/backend-common                        0.18.3
  @backstage/backend-dev-utils                     0.1.1
  @backstage/backend-plugin-api                    0.5.0
  @backstage/backend-tasks                         0.5.0
  @backstage/catalog-client                        1.4.0
  @backstage/catalog-model                         1.2.0, 1.2.1
  @backstage/cli-common                            0.1.12
  @backstage/cli                                   0.22.5
  @backstage/config-loader                         1.1.9
  @backstage/config                                1.0.7
  @backstage/core-app-api                          1.6.0
  @backstage/core-components                       0.12.4, 0.12.5
  @backstage/core-plugin-api                       1.0.6, 1.5.0
  @backstage/errors                                1.1.5
  @backstage/eslint-plugin                         0.1.2
  @backstage/integration-aws-node                  0.1.2
  @backstage/integration-react                     1.1.11
  @backstage/integration                           1.4.3
  @backstage/plugin-analytics-module-ga            0.1.27
  @backstage/plugin-api-docs                       0.9.1
  @backstage/plugin-app-backend                    0.3.43
  @backstage/plugin-auth-backend                   0.18.1
  @backstage/plugin-auth-node                      0.2.12
  @backstage/plugin-azure-devops-backend           0.3.22
  @backstage/plugin-azure-devops-common            0.3.0
  @backstage/plugin-azure-devops                   0.2.7
  @backstage/plugin-catalog-backend-module-azure   0.1.14
  @backstage/plugin-catalog-backend                1.8.0
  @backstage/plugin-catalog-common                 1.0.12
  @backstage/plugin-catalog-graph                  0.2.28
  @backstage/plugin-catalog-import                 0.9.6
  @backstage/plugin-catalog-node                   1.3.4
  @backstage/plugin-catalog-react                  1.2.1, 1.4.0
  @backstage/plugin-catalog                        1.9.0
  @backstage/plugin-github-actions                 0.5.16
  @backstage/plugin-home                           0.4.32
  @backstage/plugin-org                            0.6.6
  @backstage/plugin-permission-common              0.7.4
  @backstage/plugin-permission-node                0.7.6
  @backstage/plugin-permission-react               0.4.11
  @backstage/plugin-playlist-backend               0.2.6
  @backstage/plugin-playlist-common                0.1.5
  @backstage/plugin-playlist                       0.1.7
  @backstage/plugin-proxy-backend                  0.2.37
  @backstage/plugin-scaffolder-backend             1.12.0
  @backstage/plugin-scaffolder-common              1.2.6
  @backstage/plugin-scaffolder-node                0.1.1
  @backstage/plugin-scaffolder-react               1.2.0
  @backstage/plugin-scaffolder                     1.12.0
  @backstage/plugin-search-backend-module-pg       0.5.4
  @backstage/plugin-search-backend-node            1.1.4
  @backstage/plugin-search-backend                 1.2.4
  @backstage/plugin-search-common                  1.2.2
  @backstage/plugin-search-react                   1.5.1
  @backstage/plugin-search                         1.1.1
  @backstage/plugin-shortcuts                      0.3.8
  @backstage/plugin-sonarqube-backend              0.1.8
  @backstage/plugin-sonarqube-react                0.1.4
  @backstage/plugin-sonarqube                      0.6.5
  @backstage/plugin-tech-radar                     0.6.2
  @backstage/plugin-techdocs-backend               1.6.0
  @backstage/plugin-techdocs-module-addons-contrib 1.0.11
  @backstage/plugin-techdocs-node                  1.6.0
  @backstage/plugin-techdocs-react                 1.1.4
  @backstage/plugin-techdocs                       1.6.0
  @backstage/plugin-user-settings                  0.7.1
  @backstage/plugin-vault-backend                  0.2.10
  @backstage/plugin-vault                          0.1.10
  @backstage/release-manifests                     0.0.9
  @backstage/test-utils                            1.2.6
  @backstage/theme                                 0.2.18
  @backstage/types                                 1.0.2
  @backstage/version-bridge                        1.0.3

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else

**Comment by awanlin** [comment]
Hi @antoniobergas, I haven't looked at the code but my gut tells me it's the fact that you have both the `project-repo` and `build-definition` annotations. 

Can you try setting it up as per the documentation: https://github.com/backstage/backstage/tree/master/plugins/azure-devops#azure-pipelines-only. So you would **only** have these annotations:

```
dev.azure.com/project: <project-name>
dev.azure.com/build-definition: <build-definition-name>
```

When we added support for this it was for a use cases of someone who was only using Azure Pipelines. To make it work for you I think there will need to be some refactoring or adding support for the `dev.azure.com/project` in more places.

**Comment by antoniobergas** [comment]
That make sense! Let me try it an I let you know 😊

**Comment by antoniobergas** [comment]
I tested removing the -repo from the project annotation but when I go to the CI/CD there's no call triggering.

![image](https://user-images.githubusercontent.com/26031467/228540786-ac067d3d-8d67-40fc-b9e9-fc985a6a1aad.png)
![image](https://user-images.githubusercontent.com/26031467/228540984-505c0863-b2a2-4172-9306-85173ad39316.png)

Also tried removing quote from the build-definition and with the project name all in lower but still not working.

I am missing anything else to make work the call of the build-definitions?

**Comment by antoniobergas** [comment]
Hi @awanlin , if I understood correctly then it's need to make some changes to make this happen right?

I have no expertise with the azure-devops-node api but I can make a try if  you can guide me 😊

**Comment by awanlin** [comment]
Sorry @antoniobergas, I never saw your follow up until today. I just tested this and it works for me. Now looking at your screen shot I think something else is wrong as you should see an empty table, unless you removed that?

Can you check to see if you are using the `isAzureDevOpsAvailable` function in your `EntityPage.tsx`? I think that might be why we don't see an empty table. If you removed that and use the two annotations from my original reply does that work? As I mentioned it does for me.

**Comment by antoniobergas** [comment]
Yes! That was actually the issue. After remove the EntitySwitch surrounding the EntityAzurePipelinesContent worked at the first attempt.

Looks like the other entities with the other kind of annotation for azure repo are also fine removing the switch.

If you have any other things I should consider just tell me if not then thanks a lot for the help!!. This will make happy our frontend devs with nx monorepos 😁

**Comment by awanlin** [comment]
Awesome, glad you were able to get it working!

**Comment by subhanshu-shukla-ril** [comment]
How to make this plugin for on premises azure devops. My azure devops URL is devops.mycompany.com

**Comment by awanlin** [comment]
Hi @subhanshu-shukla-ril, this is a very old issue and this plugin no longer lives in this repo. It might be better to ask on the Backstage Discord server, I'd also look at the docs here: https://backstage.io/docs/integrations/azure/locations/
